grammar.js

```diff
@@ -374,6 +376,25 @@ module.exports = grammar({
       optional($._return_type)
     ),
 
+    arrow_function: $ => seq(
+      $._arrow_function_header,
+      field('body', $.expression_statement)
+    ),
+
+    _anonymous_arrow_function: $ => seq(
+      $._arrow_function_header,
+      field('body', alias($._expression, $.expression_statement))
+    ),
```

Comment:
Can you explain what this `_anonymous_arrow_function` is for? It seems like the same thing as `arrow_function`, except that there's no trailing `semicolon` in the body, because the body is only an `expression`, not an `expression_statement`. It makes sense that this creates a fair amount of ambiguity e.g. in cases where an arrow function is followed by a semicolon.

Just for my understanding - why not just have the `body` of `arrow_function` be an `expression` (like `_anonymous_arrow_function` is now), and remove `_anonymous_arrow_function` entirely? Could you give an example of code that wouldn't parse correctly with that structure?

Reply:
Looking at the RFC, I think arrow function bodies should always be expressions (not expression statements). I think we just need one `arrow_function` rule. Let me know if there's something I'm  missing.

Reply:
Good questions!

Adding both of them was the only way I could find to make the parser handle ; correctly in the different cases in the beginning. However I should have revisited that before thinking the PR done, cause you're perfectly correct - they should always just be expression.

Thank you for the guidance and sorry for not solving that before bugging you :)

Code updated